PostHog/PostHogSDK.swift

```diff
@@ -1013,25 +997,91 @@ let maxRetryDelay = 30.0
                 flagCallReported.removeAll()
             }
             context = nil
-            PostHogSessionManager.shared.endSession {
-                self.resetViews()
-            }
+            PostHogSessionManager.shared.endSession()
             unregisterNotifications()
             capturedAppInstalled = false
-            appFromBackground = false
-            isInBackground = false
             toggleHedgeLog(false)
             shouldReloadFlagsForTesting = true
         }
     }
 
+    #if os(iOS)
+        /**
+         Starts session recording.
+         This method will have no effect if PostHog is not enabled or if session replay integration is not available.
+
+         ## Note:
+         - This will resume the current session or create a new one if it doesn't exist
+         */
+        @objc(startSessionRecording)
+        public func startSessionRecording() {
+            startSessionRecording(resumeCurrent: true)
+        }
+
+        /**
+         Starts session recording.
+         This method will have no effect if PostHog is not enabled or if session replay integration is not available.
+         - Parameter resumeCurrent:
+            Whether to resume recording of current session (true) or start a new session (false).
+         */
+        @objc(startSessionRecordingWithResumeCurrent:)
+        public func startSessionRecording(resumeCurrent: Bool) {
+            if !isEnabled() {
```

Comment:
Should we check `isOptOutState` as well?
If we allow it as it is now and snapshots are dropped because `isOptOutState==true`, and later the user opts in, a lot of snapshots will be missing and the session will be broken, so maybe its better to not even start the session in this case?

Reply:
It's probably a use case I did not think of before, maybe makes more sense to raise a new issue with the problem, so we can address it separately, maybe the same problem happens on the other mobile platforms.